src/dataflow/core/sexp.py

```diff
@@ -149,4 +150,16 @@ def sexp_to_str(sexp: Sexp) -> str:
         else:
             return "(" + " ".join(sexp_to_str(f) for f in sexp) + ")"
     else:
-        return sexp
+        if sexp.startswith('"') and sexp.endswith('"'):
+            return sexp
+        else:
+            return _escape_symbol(sexp)
+
+
+def _escape_symbol(symbol: str) -> str:
```

Comment:
Adds escaping for arbitrary symbols to make the parser handle the same cases as the Scala parser.